### PR TITLE
Add NNCF quantization for Whisper notebook

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Work in progress

Ticket: CVS-112132

### CODE REVIEW

**Comment by raymondlo84** [comment]
Wow this is so cool! @adrianboguszewski

File: notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb [hunk 1]
```diff
@@ -0,0 +1,941 @@
+{
```
Comment (ngaloppo) [inline notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    #!pip install -q evaluate
</pre><p> </p><p>I think this notebook is missing dependencies: <code>librosa, soundfile, jiwer</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1237/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (nikita-savelyevv) [reply notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb:1]: Thanks!

File: notebooks/227-whisper-subtitles-generation/README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ In this notebook, you will use its capabilities for generation of subtitles for
 Notebook contains the following steps:
```
Comment (MaximProshin) [inline notebooks/227-whisper-subtitles-generation/README.md:13]: Update the description that there are 2 tutorials: how to convert and how to optimize Whisper.

File: notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb [hunk 3]
```diff
@@ -0,0 +1,1167 @@
...
+    "from openvino.runtime import serialize\n",
+    "\n",
+    "print(\"Quantizing encoder...\")\n",
+    "quantized_encoder = nncf.quantize(\n",
+    "    model=model_fp32.encoder.model,\n",
+    "    calibration_dataset=nncf.Dataset(encoder_calibration_data),\n",
+    "    subset_size=len(encoder_calibration_data),\n",
+    "    model_type=nncf.ModelType.TRANSFORMER,\n",
+    "    advanced_parameters=nncf.AdvancedQuantizationParameters(\n",
+    "        overflow_fix=OverflowFix.DISABLE,\n",
```
Comment (MaximProshin) [inline notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb:522]: please add comment why it's needed

File: notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb [hunk 4]
```diff
@@ -0,0 +1,1167 @@
...
+    "        smooth_quant_alpha=0.5\n",
```
Comment (MaximProshin) [inline notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb:523]: please add a comment why it's needed

File: .ci/spellcheck/.pyspelling.wordlist.txt [hunk 5]
```diff
@@ -550,4 +550,6 @@ YOLOv
 yolov
 Zongyuan
 ZeroScope
-zeroscope
\ No newline at end of file
+zeroscope
+librispeech
```
Comment (eaidova) [inline .ci/spellcheck/.pyspelling.wordlist.txt:554]: please keep alphabetical order adding words in wordlist

File: notebooks/227-whisper-subtitles-generation/README.md (hunk 2 above)
Comment (nikita-savelyevv) [reply notebooks/227-whisper-subtitles-generation/README.md]: Done

File: notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb (hunk 3 above)
Comment (nikita-savelyevv) [reply notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb]: Done

File: notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb (hunk 4 above)
Comment (nikita-savelyevv) [reply notebooks/227-whisper-subtitles-generation/227-whisper-nncf-quantize.ipynb]: Done

File: .ci/spellcheck/.pyspelling.wordlist.txt (hunk 5 above)
Comment (nikita-savelyevv) [reply .ci/spellcheck/.pyspelling.wordlist.txt]: Done

**Review by eaidova — APPROVED** [review]

**Review by MaximProshin — APPROVED** [review]
LGTM
I also executed it on my i7 laptop and got 1.7x boost for encoder and 2.4x for decoder with int8.

**Pull request merged**